src/Interpreters/Session.cpp

```diff
@@ -29,6 +29,7 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
     extern const int SESSION_NOT_FOUND;
     extern const int SESSION_IS_LOCKED;
+    extern const int AUTHENTICATION_DENIED;
```

Comment:
We already have `AUTHENTICATION_FAILED`, I think we don't need to introduce another similar error code.